raiden-ts/src/services/epics/udc.ts

```diff
@@ -88,27 +88,36 @@ function makeUdcDeposit$(
         tokenContract.callStatic.allowance(sender, userDepositContract.address) as Promise<
           UInt<32>
         >,
-        userDepositContract.callStatic.total_deposit(address),
       ]),
     pollingInterval,
     { onErrors: networkErrors },
   ).pipe(
-    mergeMap(([balance, allowance, deposited]) => {
-      assert(deposited.add(deposit).eq(totalDeposit), [
-        ErrorCodes.UDC_DEPOSIT_OUTDATED,
-        { requested: totalDeposit.toString(), current: deposited.toString() },
-      ]);
-      return approveIfNeeded$(
+    mergeMap(([balance, allowance]) =>
+      approveIfNeeded$(
         [balance, allowance, deposit],
         tokenContract,
         userDepositContract.address as Address,
         ErrorCodes.RDN_APPROVE_TRANSACTION_FAILED,
         { provider },
         { log, minimumAllowance },
-      );
-    }),
-    // send setTotalDeposit transaction
-    mergeMap(() => userDepositContract.deposit(address, totalDeposit)),
+      ),
+    ),
+    mergeMap(() =>
+      retryAsync$(
+        async () => userDepositContract.callStatic.total_deposit(address),
+        pollingInterval,
+        { onErrors: networkErrors },
+      ).pipe(
+        mergeMap(async (deposited) => {
+          assert(deposited.add(deposit).eq(totalDeposit), [
+            ErrorCodes.UDC_DEPOSIT_OUTDATED,
+            { requested: totalDeposit.toString(), current: deposited.toString() },
+          ]);
```

Comment:
Codeclimate was failing, complaining about this because it was skipped before, and since I've split and moved this code around, it triggered again. It also used some outdated patterns on the retry, so it was reasonable to adapt and improve those while making Codeclimate happy.